manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -113,9 +113,20 @@ type ResourceReference struct {
 // DatasetDetails holds details of the provisioned storage
 type DatasetDetails struct {
 	// Reference to a Dataset resource containing the request to provision storage
+	// +optional
 	DatasetRef string `json:"datasetRef,omitempty"`
-	// Reference to a secret where the credentials are stored
-	SecretRef string `json:"secretRef,omitempty"`
+
+	// Name of the secret where the credentials are stored
+	// +optional
+	SecretName string `json:"secretName,omitempty"`
+
+	// Name of the secret namespace where the credentials are stored
+	// +optional
+	SecretNamespace string `json:"secretNamespace,omitempty"`
```

Comment:
Create a SecretRef structure and put both there

Reply:
It appears that `type NamespacedName` can not be used as is because it is missing JSON tag so I will create a new one.